### PR TITLE
Change README.md CI badges to use GitHub actions

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
## Description

Addresses #3223 

The good news: GitHub has a lot of documentation for adding and customising badges
The bad news: There isn't (yet) a way to customise the text on the badge [xref](https://github.com/orgs/community/discussions/63964). Currently using the same third party customiser that we used for the CircleCI badges.

The badges check the `all-checks.yml` GH Actions workflow for branches main and develop, respectively (GH Actions badges are workflow specific). 

## Development notes

[Link to the README with new badges](https://github.com/kedro-org/kedro/blob/ci/add-gh-actions-badge-to-readme/README.md)

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by ankatiyar** [comment]
I think shields.io also has badges for Github Actions - https://shields.io/badges/git-hub-actions-workflow-status you can try that. 
Also, there's badges on the docs front page as well which can be changed!

**Comment by AhdraMeraliQB** [comment]
> I think shields.io also has badges for Github Actions - https://shields.io/badges/git-hub-actions-workflow-status you can try that. Also, there's badges on the docs front page as well which can be changed!

Noticed the shields.io badges, and good spot for the docs!

File: docs/source/index.rst [hunk 1]
```diff
@@ -11,13 +11,13 @@
 Welcome to Kedro's documentation!
```
Comment (stichbury) [inline docs/source/index.rst:11]:
```suggestion
Welcome to Kedro's award-winning documentation!
```

Comment (stichbury) [reply docs/source/index.rst:11]: While you're in there making changes, are you OK to commit this as part of the set of changes?

**Comment by stichbury** [comment]
Checked on the docs build https://kedro--3501.org.readthedocs.build/en/3501/ and all looks good to me (the links work fine). Thank you!

**Review by stichbury — APPROVED** [review]

File: docs/source/index.rst (hunk 1 above)
Comment (AhdraMeraliQB) [reply docs/source/index.rst:11]: Absolutely!

**Review by ankatiyar — APPROVED** [review]
Looks good to me! ⭐

**Pull request merged**